tracks/python/exercises/say/mentoring.md

```diff
@@ -0,0 +1,127 @@
+# Mentoring
+
+## Problems and Challenges
+This problem requires the student to transform an integer (0 <= n < 1,000,000,000,000) into its full English equivalent, and raise an exception for values outside that range.
+Take note of the tests - the use of 'and' as a separator is not required, and adding it will cause the tests to fail, but it is suggested as an extension exercise. 
+
+## Reasonable Solution
+```
+digits = [""] + "one two three four five six seven eight nine".split()
+teens = "ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen".split()
+tens = ["", ""] + "twenty thirty forty fifty sixty seventy eighty ninety".split()
+denominations = (9, "billion"), (6, "million"), (3, "thousand"), (2, "hundred")
+
+def say(number):
+    if 0 > number or number > 999999999999:
+        raise ValueError("input out of range")
+    if number < 20 and number >= 10:
+        return teens[number % 10]
+    if number < 100:
+        if number // 10 and number % 10:
+            return "-".join([tens[number // 10], digits[number % 10]])
+        else:
+            return tens[number // 10] or digits[number % 10] or "zero"
+    for k, denomination in denominations:
+        if number > 10**k - 1:
+            result = [say(number // 10**k), denomination]
+            if number % 10 ** k:
+                result.append(say(number % 10**k))
+            return " ".join(result)
+```
+Or without using recursion:
+```
+def say3(number):
```

Comment:
`say3` isn't a very description function name. Maybe `say_small`?

Reply:
I've changed it to `say_three_digits`